cgi/product_image_upload.pl

```diff
@@ -313,7 +313,9 @@
 			# Changed 2020-04-20: don't overwrite selected images if the source is the product edit form
 			and (  (not defined single_param('source'))
 				or (single_param('source') ne "product_edit_form")
-				or (not defined $product_ref->{images}{$imagefield}))
+				or (not defined $product_ref->{images}{$imagefield}) and ($User{moderator})
```

Comment:
There's an issue with the conditions here. We should keep all the previous conditions, and in addition to that, we need to check that the image is not protected (unless the user is the moderator)